### DISCUSSION TITLE
Deploying a database for each preview deployment (how manage order, and set ENV vars?)

### DISCUSSION THREAD

**Discussion by einarpersson** [vercel/community#1523] [Help]
### Summary

Hi!

My goal is to have a temporary database instance accompanying each PR preview deployment so that I can reliably run Playwright e2e-tests each time a PR is updated. I'm using Sveltekit and the database is Pockedbase, deployed with a Dockerfile to Fly.io.

This means
A) The vercel preview deployment has to have _some way of knowing the URL of the database_ (set env variable at build time or just rely on some deterministic formula for building the db URL from branch name)
B) The vercel preview _deployment build should preferably not start until database is up and running_, since content in database might be used during build for pre-rendering. If this increases the complexity a lot, I might consider skipping this for now, but it would be good to know if/how it could be achieved
C) Github also needs to be _notified when a deployment is done so that e2e-tests can start_

I am not sure whether I should try to rely on the automatic github integration and the Sveltekit vercel-adapter or if I have to use the Vercel CLI directly for achieving this. What is the relation between vercel-adapter, vercel CLI, vite, ..?

1. Given my requirements above, what approach would you recommend?
2. If I use vercel CLI to deploy, will an event still be sent to github for triggering workflow? (if I have also connected the project to the repo, but turned off automatic deploys as `git.deploymentEnabled: false` in `vercel.json`)
3. Or is it maybe an idea to just prepend a "wait for db healthcheck" as a part of the deploy command that vercel runs? `npm run db-check && npm run build`. Now that I think of it, maybe this would be the easiest way forward.

**Comment by amyegan** [comment]
Adding the db health check to your deploy command is probably the easiest option to try first, since you only need to change one line. But I'm not sure how well it would work in your case. It depends on how you kick off temporary db deployment and a bunch of other factors. This does seem like the simplest place to start if you haven't already tried it.

If you want an more customizable automatic process to kick off both the db and site deployments, you'll get the most control with GitHub Actions + Vercel CLI. We have a guide on [using GitHub Actions with Vercel](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) that includes a good starter script.

**Reply by einarpersson** [reply]
Thanks for getting back to me @amyegan. I prepended a `db-check.js`-script which worked, but I woke up this morning realizing that this is faulty thinking anyway. The database might respond "healthy" and allow the build to move forward, but that won't guarantee that migrations have had the time to run. I see no other way but to create a custom Github workflow for a predictable and orderly behaviour. I will try to follow the instruction in the link you provided. 

Moving forward, I would be very grateful if you could also answer the following:

1. Should I set `git.deploymentEnabled: false` in `vercel.json`?
2. Will doing so disable `deployment_status` events from being posted to github on deploy or will the event still get sent on deploy, even if it is done manually using the Vercel CLI (as long as the Vercel project is connected to the repository)? 
3. What is the relationship between using `vercel build` and the framework specific `vite build`? Which one should I use and what happens under the hood?  Will `vercel build` simply run the build command specified for the Vercel project, with Vercel environment variables available? (then again, `vite build` will run with sveltekit as a plugin, which in turn has something called a `vercel-adapter` configured, so it is very confusing how everything fits together 😅)

If I get this to work I might write a blogpost about it:)
